doc/background/custom_executors.adoc

```diff
@@ -0,0 +1,151 @@
+== Custom Executors
+
+One of the reasons async defaults to https://www.boost.org/doc/libs/1_83_0/doc/html/boost_asio/reference/any_io_executor.html::[`asio::any_io_executor`]
+is that it is a type-erased executor, i.e. you can provide your own event-loop without needing to recompile `async`.
+
+However, during the development of the Executor TS, the executor concepts got a bit unintuitive, to put it mildly.
+
+Ruben Perez wrote an excellent https://anarthal.github.io/cppblog/asio-props.html::[blog post], which I am shamelessly going to draw from.
+
+=== Definition
+
+An executor is a type that points to the actual event loop and is (cheaply) copyable,
+which supports properties (see below) is equality comparable and has an `execute` function.
+
+==== `execute`
+
+[source,cpp]
+----
+struct example_executor
+{
+  template<typename Fn>
+  void executor(Fn && fn) const;
+};
+----
+
+The above function executes `fn` in accordance with its properties.
+
+==== Properties
+
+A property can be queried, preferred or required, e.g.:
+
+[source,cpp]
+----
+struct example_executor
+{
+  // get a property by querying it.
+  asio::execution::relationship_t &query(asio::execution::relationship_t) const
+  {
+    return asio::execution::relationship.fork;
+  }
+
+  // require an executor with a new property
+  never_blocking_executor require(const execution::blocking_t::never_t);
+
+  // prefer an executor with a new property. the executor may or may not support it.
+  never_blocking_executor prefer(const execution::blocking_t::never_t);
+  // not supported
+  example_executor prefer(const execution::blocking_t::always_t);
+};
+----
+
+==== Properties of the `asio::any_io_executor`
+
+In order to wrap an executor in an `asio::any_io_executor` two properties are required:
```

Comment:
I think this is incomplete, looking at the source code:

      execution::context_as_t<execution_context&>,
      execution::blocking_t::never_t,
      execution::prefer_only<execution::blocking_t::possibly_t>,
      execution::prefer_only<execution::outstanding_work_t::tracked_t>,
      execution::prefer_only<execution::outstanding_work_t::untracked_t>,
      execution::prefer_only<execution::relationship_t::fork_t>,
      execution::prefer_only<execution::relationship_t::continuation_t>

Reply:
The `prefer_only`s are required?